### PR TITLE
Rajatvarma master

### PR DESCRIPTION
**Pull request by ViRu-ThE-ViRuS**
@rajatvarma commits
Had to do this to clean up the merge conflicts
Param review and merge pls.

### CODE REVIEW

File: QuestionSets/PythonLanguage/ClassesAndObjects/Rajat_Varma/Enemy.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        self.health -= 50
+
+    def heal(self):
+        self.health = 100
+
+    def booster(self):
+        self.attack_power += 50
+
+    def attack(self):
+        Player.take_damage(Jason_Brody)
```
Comment (ViRu-ThE-ViRuS) [inline QuestionSets/PythonLanguage/ClassesAndObjects/Rajat_Varma/Enemy.py]: @rajatvarma Is `Jason_Brody` (love the Far Cry Reference tho) really taking damage ie. is his health decreasing??

Comment (rajatvarma) [reply QuestionSets/PythonLanguage/ClassesAndObjects/Rajat_Varma/Enemy.py]: Yes. If I print his health at the end of Enemy.py, I get 90.

**Comment by paramkpr** [comment]
@ViRu-ThE-ViRuS Thanks. Merge Errors suck.

**Comment by ViRu-ThE-ViRuS** [comment]
@paramkpr abe phir se bina check kar merge kar lia.... the program is not correct!

File: QuestionSets/PythonLanguage/ClassesAndObjects/Rajat_Varma/Enemy.py (hunk 1 above)
Comment (ViRu-ThE-ViRuS) [reply QuestionSets/PythonLanguage/ClassesAndObjects/Rajat_Varma/Enemy.py]: man, this code is wrong.... ill tell u why in the next session, just remind me to

**Comment by paramkpr** [comment]
@ViRu-ThE-ViRuS The code works fine... what exactly is the issue?

**Comment by ViRu-ThE-ViRuS** [comment]
@paramkpr @rajatvarma  not every python code which runs is "pythonic". Check the line i have reviewed above... this is not how its meant to be done. There is much more to code than whether it just runs or not. 

#### Moral:
This is not the appropriate way to solve this problem. It can lead to some untraceable issues while debugging.

#### Details:
- `take_damage` method is  being called as if it were a static method, but its not annotated to be such. You are supposed to call the `take_damage` method of `Jason_Brody` not pass `Jason_Brody` as a parameter to the function (if it were so, the respective type annotation should have been used `@staticmethod`).
- The reason this code works is different, its cause of the design of python.

I will make a pr and correct this, this time.

**Pull request merged**